HPCC-21496 Couchbase query requires credentials

rpastrana opened the pull request:
- Updates libcouchbase-cxx submodule to latest from master
- Adds support for couchbase bucket user name
- Adds support for documented connection options
- Updates sample ecl files to utilize user/pass vars

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

rpastrana commented:
@AttilaVamos this change should fix the smoketest issue, please help test it

HPCCSmoketest commented:
Process of PR-12565, label: HPCC-21496couchbasecreds starts now.
The reason of this test is: Did not build yet.
Commit ID: d020d98f0a1a12e6a069da0fe1f32f5d86ce319f
Estimated completion time is ~0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: d020d98f0a1a12e6a069da0fe1f32f5d86ce319f
Number of Error(s): 5
/mnt/disk1/home/vamosax/smoketest/PR-12565/HPCC-Platform/plugins/couchbase/couchbaseembed.hpp:194:94: error: no matching function for call to \xe2\x80\x98Couchbase::Client::Client(const char*, const char*&, const char*&)\xe2\x80\x99
/mnt/disk1/home/vamosax/smoketest/PR-12565/HPCC-Platform/plugins/couchbase/couchbaseembed.hpp:200:94: error: no matching function for call to \xe2\x80\x98Couchbase::Client::Client(const char*, const char*&, const char*&)\xe2\x80\x99
make[2]: *** [plugins/couchbase/CMakeFiles/couchbaseembed.dir/couchbaseembed.cpp.o] Error 1
make[1]: *** [plugins/couchbase/CMakeFiles/couchbaseembed.dir/all] Error 2
make: *** [all] Error 2

[Build error:](https://gist.github.com/5ab9fbba1d5b945d62c37704984ea6ce)
Build log [Build log](https://gist.github.com/HPCCSmoketest/5ab9fbba1d5b945d62c37704984ea6ce#file-result-19-05-20-10-26-37-log)

rpastrana commented:
@AttilaVamos I believe this is failing b/c the submodule was not updated as expected, I'll follow up with @richardkchapman

rpastrana committed:
HPCC-21496 Couchbase query requires credentials

- Updates libcouchbase-cxx submodule to latest from master
- Adds support for couchbase bucket user name
- Adds support for documented connection options
- Updates sample ecl files to utilize user/pass vars

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

HPCCSmoketest commented:
Process of PR-12565, label: HPCC-21496couchbasecreds starts now.
The reason of this test is: New commit.
Commit ID: 3cea714c41d120710332a7337739bb50d4817d22
Estimated completion time is ~0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 3cea714c41d120710332a7337739bb50d4817d22
Number of Error(s): 5
/mnt/disk1/home/vamosax/smoketest/PR-12565/HPCC-Platform/plugins/couchbase/couchbaseembed.hpp:194:94: error: no matching function for call to \xe2\x80\x98Couchbase::Client::Client(const char*, const char*&, const char*&)\xe2\x80\x99
/mnt/disk1/home/vamosax/smoketest/PR-12565/HPCC-Platform/plugins/couchbase/couchbaseembed.hpp:200:94: error: no matching function for call to \xe2\x80\x98Couchbase::Client::Client(const char*, const char*&, const char*&)\xe2\x80\x99
make[2]: *** [plugins/couchbase/CMakeFiles/couchbaseembed.dir/couchbaseembed.cpp.o] Error 1
make[1]: *** [plugins/couchbase/CMakeFiles/couchbaseembed.dir/all] Error 2
make: *** [all] Error 2

[Build error:](https://gist.github.com/774c15ff44d1a6185931d864d46b2ea6)
Build log [Build log](https://gist.github.com/HPCCSmoketest/774c15ff44d1a6185931d864d46b2ea6#file-result-19-05-20-10-56-45-log)

HPCCSmoketest commented:
Process of PR-12565, label: HPCC-21496couchbasecreds starts now.
The reason of this test is: New commit.
Commit ID: e265a0002009d77847820ca5a1650cf404481cba
Estimated completion time is ~0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: e265a0002009d77847820ca5a1650cf404481cba
Number of Error(s): 5
/mnt/disk1/home/vamosax/smoketest/PR-12565/HPCC-Platform/plugins/couchbase/couchbaseembed.hpp:194:94: error: no matching function for call to \xe2\x80\x98Couchbase::Client::Client(const char*, const char*&, const char*&)\xe2\x80\x99
/mnt/disk1/home/vamosax/smoketest/PR-12565/HPCC-Platform/plugins/couchbase/couchbaseembed.hpp:200:94: error: no matching function for call to \xe2\x80\x98Couchbase::Client::Client(const char*, const char*&, const char*&)\xe2\x80\x99
make[2]: *** [plugins/couchbase/CMakeFiles/couchbaseembed.dir/couchbaseembed.cpp.o] Error 1
make[1]: *** [plugins/couchbase/CMakeFiles/couchbaseembed.dir/all] Error 2
make: *** [all] Error 2

[Build error:](https://gist.github.com/0f52ff333e1993a6e71bf8fa5a9c1623)
Build log [Build log](https://gist.github.com/HPCCSmoketest/0f52ff333e1993a6e71bf8fa5a9c1623#file-result-19-05-20-14-27-40-log)

HPCCSmoketest commented:
Process of PR-12565, label: HPCC-21496couchbasecreds starts now.
The reason of this test is: New commit.
Commit ID: e265a0002009d77847820ca5a1650cf404481cba
Estimated completion time is ~0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: e265a0002009d77847820ca5a1650cf404481cba
Build: success
Build: success
Install HPCC Platform
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |113 |113 |0 |0 |0 |36 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |24 sec  (00:00:24) |
|test (hthor) |829 |829 |0 |164 sec  (00:02:44) |
|test (thor) |754 |754 |0 |602 sec  (00:10:02) |
|test (roxie) |905 |905 |0 |218 sec  (00:03:38) |


HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   29 sec (00:00:29) |  214 sec (00:03:34) |    0 sec (00:00:00) |    2 sec (00:00:02) |   19 sec (00:00:19) | 1299 sec (00:21:39) |   20 sec (00:00:20) | 1583 sec (00:26:23) |

rpastrana commented:
confirmed fix on 160.

rpastrana commented:
@richardkchapman please review

richardkchapman commented:
If we require a change to the submoduled library, then this should be a 7.4 change

HPCCSmoketest commented:
Process of PR-12565, label: HPCC-21496couchbasecreds starts now.
The reason of this test is: Base branch changed.
Commit ID: e265a0002009d77847820ca5a1650cf404481cba
Estimated completion time is ~0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: e265a0002009d77847820ca5a1650cf404481cba
Conflicting files, should skip build and test.

[Build error:](https://gist.github.com/b6e5647d95116e145ac9cbef02f48493)
Build log [Build log](https://gist.github.com/HPCCSmoketest/b6e5647d95116e145ac9cbef02f48493#file-result-19-05-23-09-09-27-log)

HPCCSmoketest commented:
Process of PR-12565, label: HPCC-21496couchbasecreds starts now.
The reason of this test is: New commit.
Commit ID: bf1eddf1ac977e921994f3b2e05d10560dcdafb7
Estimated completion time is ~0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: bf1eddf1ac977e921994f3b2e05d10560dcdafb7
Build: success
Build: success
Install HPCC Platform
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |113 |113 |0 |0 |0 |36 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |28 sec  (00:00:28) |
|setup (thor) |11 |11 |0 |46 sec  (00:00:46) |
|setup (roxie) |11 |11 |0 |24 sec  (00:00:24) |
|test (hthor) |834 |834 |0 |181 sec  (00:03:01) |
|test (thor) |759 |759 |0 |600 sec  (00:10:00) |
|test (roxie) |910 |910 |0 |232 sec  (00:03:52) |


HPCC Stop: OK

HPCCSmoketest commented:
Process of PR-12565, label: HPCC-21496couchbasecreds starts now.
The reason of this test is: New commit.
Commit ID: 504044c49d8bfcc05035a222ecef69320e4c536a
Estimated completion time is ~0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 504044c49d8bfcc05035a222ecef69320e4c536a
Build: success
Build: success
Install HPCC Platform
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |113 |113 |0 |0 |0 |35 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |7 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |26 sec  (00:00:26) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |21 sec  (00:00:21) |
|test (hthor) |834 |834 |0 |194 sec  (00:03:14) |
|test (thor) |759 |759 |0 |615 sec  (00:10:15) |
|test (roxie) |910 |910 |0 |217 sec  (00:03:37) |


HPCC Stop: OK
